src/main/java/edu/ie3/datamodel/models/input/thermal/CylindricalStorageInput.java

```diff
@@ -157,4 +155,76 @@ public String toString() {
         + c
         + '}';
   }
+
+  /**
+   * A builder pattern based approach to create copies of {@link CylindricalStorageInput} entities
+   * with altered field values. For detailed field descriptions refer to java docs of {@link
+   * CylindricalStorageInput}
+   */
+  public class CylindricalStorageInputCopyBuilder
+      extends AssetInput.AssetInputCopyBuilder<CylindricalStorageInputCopyBuilder> {
+
+    private ComparableQuantity<Volume> storageVolumeLvl;
+    private ComparableQuantity<Volume> storageVolumeLvlMin;
+    private ComparableQuantity<Temperature> inletTemp;
+    private ComparableQuantity<Temperature> returnTemp;
+    private ComparableQuantity<SpecificHeatCapacity> c;
+
+    private CylindricalStorageInputCopyBuilder(CylindricalStorageInput entity) {
+      super(entity);
+      this.storageVolumeLvl = entity.getStorageVolumeLvl();
+      this.storageVolumeLvlMin = entity.getStorageVolumeLvlMin();
+      this.inletTemp = entity.getInletTemp();
+      this.returnTemp = entity.getReturnTemp();
+      this.c = entity.getC();
+    }
+
+    @Override
+    public CylindricalStorageInput build() {
+      return new CylindricalStorageInput(
+          getUuid(),
+          getId(),
+          getOperator(),
+          getOperationTime(),
+          getThermalBus(),
```

Comment:
You are pointing to the wrong thermal bus, here.

As your inner class is not static, you are able to point to the `CylindricalStorageInput#getThermalBus`, so more or less to the instance you wanna copy. Meaning, at the moment you are not able to alter the bus within the copy builder. You would need to point to a thermal bus within the copy builder-hierarchy.

As the thermal bus is used by several other classes as well, I would suggest to implement a new *abstract* class `ThermalUnitInpuiutCopyBuilder`  extending `AssetInputCopyBuilder` as inner class of `ThermalUnitInput`. It only holds a thermal bus as attribute and adds equivalent methods to set and get it in the same style as `AssetInputCopyBuilder`.

Then `CylindricalStorageInputCopyBuilder` and `ThermalHouseInputCopyBuilder` can extend the `ThermalUnitInputCopyBuilder`.

Reply:
Sure, will make the changes.